File: seoyeon/week11/명예의 전당(1).py
def solution(k, score):
    answer = [] # 최하위 점수 배열
    array = []
    for i in score :
        if len(array) < k :
            array.append(i)
        elif min(array) < i :
            array.remove(min(array))
            array.append(i)
        array.sort(reverse=True)
        m = array[-1]
        answer.append(m)
    return answer

File: hyeyoung/week11/명예의전당(1).java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int min = 0;
        List<Integer> list = new ArrayList<>();
        
        for(int i = 0; i< score.length; i++){
            if(list.size() < k){
                list.add(score[i]);
                Collections.sort(list);
                answer[i] = list.get(0);
            }else if(list.size() == k){
                min = list.get(0);
                if(min < score[i]){
                    list.remove(0);
                    list.add(score[i]);
                    Collections.sort(list);
                }
                answer[i] = list.get(0);
            }
        }
        return answer;
    }
}
